src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -6,33 +6,41 @@
 using System.IO;
 using System.Linq;
 using NuGet.ProjectModel;
+using NuGet.Packaging.Core;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
 
 namespace Microsoft.NET.Build.Tasks
 {
     public class ProjectContext
     {
         private readonly LockFile _lockFile;
-        private readonly LockFileTarget _filterlockFileTarget;
         private readonly LockFileTarget _lockFileTarget;
         private readonly string _platformLibraryName;
+        private HashSet<PackageIdentity> _packagesToBeFiltered;
 
         public bool IsPortable { get; }
         public LockFileTargetLibrary PlatformLibrary { get; }
 
         public LockFile LockFile => _lockFile;
         public LockFileTarget LockFileTarget => _lockFileTarget;
 
-        public ProjectContext(LockFile lockFile, LockFileTarget lockFileTarget, string platformLibraryName, LockFileTarget filterlockFileTarget = null)
+        public ProjectContext(LockFile lockFile, LockFileTarget lockFileTarget, string platformLibraryName)
         {
             _lockFile = lockFile;
-            _filterlockFileTarget = filterlockFileTarget;
             _lockFileTarget = lockFileTarget;
             _platformLibraryName = platformLibraryName;
 
             PlatformLibrary = _lockFileTarget.GetLibrary(_platformLibraryName);
             IsPortable = PlatformLibrary != null && string.IsNullOrEmpty(_lockFileTarget.RuntimeIdentifier);
         }
 
+        internal ProjectContext AddpackagestoBeFiltered(HashSet<PackageIdentity> packagesToBeFiltered)
```

Comment:
This isn't cased correctly.

Also - why not just make this a normal property with a getter and setter?